lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -682,3 +684,42 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
   removeCall();
   return true;
 }
+
+bool swift::ArraySemanticsCall::replaceByAppendingValues(SILModule &M,
+  SILFunction *AppendFn,
+  SmallVectorImpl<SILValue> &Vs,
+  ArrayRef<Substitution> Subs) {
+  assert(getKind() == ArrayCallKind::kAppendContentsOf &&
+         "Must be an append_contentsOf call");
+  assert(AppendFn && "Must provide an append SILFunction");
+
+  // We only handle loadable types.
+  for(auto &V : Vs) {
+    if (!V->getType().isLoadable(M))
+      return false;
+  }
```

Comment:
This is not LLVM style. We avoid predicate loops in LLVM style. I would suggest this:

```
if (any_of(Vs, [&M](SILValue V) -> bool {
                  return !V.getType().isLoadable(M);
                })) {
    return false;
}
```